src/bedrock/route/routing.cpp

```diff
@@ -117,257 +101,55 @@ void run(unsigned thread_id) {
 
     // only relavant for the seed node
     if (pollitems[0].revents & ZMQ_POLLIN) {
-      logger->info("Received an address request.");
-      zmq_util::recv_string(&addr_responder);
-
-      communication::Address address;
-      address.set_start_time(duration);
-      for (auto it = global_hash_ring_map.begin(); it != global_hash_ring_map.end(); it++) {
-        unsigned tier_id = it->first;
-        auto hash_ring = &(it->second);
-        unordered_set<string> observed_ip;
-
-        for (auto iter = hash_ring->begin(); iter != hash_ring->end(); iter++) {
-          if (observed_ip.find(iter->second.get_ip()) == observed_ip.end()) {
-            communication::Address_Tuple* tp = address.add_tuple();
-            tp->set_tier_id(tier_id);
-            tp->set_ip(iter->second.get_ip());
-
-            observed_ip.insert(iter->second.get_ip());
-          }
-        }
-      }
-
-      string serialized_address;
-      address.SerializeToString(&serialized_address);
-      zmq_util::send_string(serialized_address, &addr_responder);
+      seed_handler(logger,
+                   &addr_responder,
```

Comment:
Seems clunky to be passing around ZMQ references. Can we receive call `recv_string` in `routing.cpp` and then passing in the string instead?

Reply:
This applies to all the handler calls. Not going to comment on future ones.

Reply:
I tend to think the current way is fine...the "received" msg may or may not get used in each handler (for example, the seed_handler just receive the msg without ever using it), so I think receiving inside the handler may make more sense.